Document make-channel.nix

roberth opened the pull request:
###### Motivation for this change

This documentation was written 11 months ago by @lheckemann during the NixCon 2017 hackathon. I've rebased it so it can be of use.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

lheckemann committed:
Document make-channel.nix

matthewbauer commented on nixos/lib/make-channel.nix:
```diff
@@ -1,3 +1,7 @@
+/* Build a channel tarball. These contain, in addition to the nixpkgs
+ * expressions themselves, files that indicate the version of nixpkgs
+ * that they represent. This only seems to be used by Travis.
```
Not travis but hydra

lheckemann replied:
Yes, I'm not sure where I got that idea from…

roberth replied:
You were probably listening to [Rae & Christian](https://genius.com/Rae-and-christian-travis-lyrics) ;)

lheckemann commented:
Thanks for picking this up! I'd completely forgotten about it…

roberth committed:
Remove incorrect sentence from make-channel comment

Besides being incorrect, claims like these tend to stick around
after they become incorrect.
